platform/ios/src/MGLScaleControlView.m

```diff
@@ -0,0 +1,48 @@
+#import "MGLScaleControlView.h"
+
+@interface MGLScaleControlView()
+@property (nonatomic, assign) double zoomLevel;
+@property (nonatomic, assign) CLLocationDistance metersPerPoint;
+@property (nonatomic) NSArray *labels;
+@property (nonatomic) NSArray *bars;
+@property (nonatomic) UILabel *debugLabel;
+@end
+
+@implementation MGLScaleControlView
+
+- (void)updateWithZoomLevel:(double)zoomLevel metersPerPoint:(CLLocationDistance)metersPerPoint {
+    self.zoomLevel = zoomLevel;
+    self.metersPerPoint = metersPerPoint;
+}
+
+- (void)setZoomLevel:(double)zoomLevel {
+    _zoomLevel = zoomLevel;
+    [self.debugLabel setText:[NSString stringWithFormat:@"%0.2f", zoomLevel]];
+    [self.debugLabel sizeToFit];
+    [self setNeedsLayout];
+}
+
+- (CGSize)intrinsicContentSize {
+    return CGSizeMake(100, 10);
+}
+
+- (UILabel *)debugLabel {
+    if (!_debugLabel) {
+        _debugLabel = [[UILabel alloc] init];
+        _debugLabel.font = [UIFont systemFontOfSize:8];
+        [self addSubview:_debugLabel];
```

Comment:
A label means you'd have to manage constraints internal to this view. That may be a good idea, but consider whether it might be easier to override -drawInRect: and use Core Graphics and attributed string drawing methods to accomplish the same effect, since you don't need to worry about line wrapping.